sysbuild/CMakeLists.txt

```diff
@@ -241,16 +241,48 @@ function(${SYSBUILD_CURRENT_MODULE_NAME}_pre_cmake)
 
     if(SB_CONFIG_PARTITION_MANAGER)
       # Use NCS signing script with support for PM
-      set(${DEFAULT_IMAGE}_SIGNING_SCRIPT "${ZEPHYR_NRF_MODULE_DIR}/cmake/sysbuild/image_signing.cmake" CACHE INTERNAL "MCUboot signing script" FORCE)
+      if(SB_CONFIG_QSPI_XIP_SPLIT_IMAGE)
+        set(${DEFAULT_IMAGE}_SIGNING_SCRIPT "${ZEPHYR_NRF_MODULE_DIR}/cmake/sysbuild/image_signing_split.cmake" CACHE INTERNAL "MCUboot signing script" FORCE)
 
-      if(SB_CONFIG_MCUBOOT_BUILD_DIRECT_XIP_VARIANT)
-        set(mcuboot_secondary_app_SIGNING_SCRIPT "${ZEPHYR_NRF_MODULE_DIR}/cmake/sysbuild/image_signing.cmake" CACHE INTERNAL "MCUboot signing script" FORCE)
-      elseif(SB_CONFIG_MCUBOOT_MODE_FIRMWARE_UPDATER)
-        set(firmware_loader_SIGNING_SCRIPT "${ZEPHYR_NRF_MODULE_DIR}/cmake/sysbuild/image_signing_firmware_loader.cmake" CACHE INTERNAL "MCUboot signing script" FORCE)
+        if(SB_CONFIG_MCUBOOT_BUILD_DIRECT_XIP_VARIANT)
+          set(mcuboot_secondary_app_SIGNING_SCRIPT "${ZEPHYR_NRF_MODULE_DIR}/cmake/sysbuild/image_signing_split.cmake" CACHE INTERNAL "MCUboot signing script" FORCE)
+        endif()
 
-        set_target_properties(firmware_loader PROPERTIES
-          IMAGE_CONF_SCRIPT ${ZEPHYR_NRF_MODULE_DIR}/sysbuild/image_configurations/firmware_loader_image_default.cmake
-        )
+        # Share QSPI XIP MCUboot image number with the default image to allow for signing script
+        # to get the partition manager configuration
+        set(${DEFAULT_IMAGE}_QSPI_XIP_IMAGE_NUMBER ${SB_CONFIG_MCUBOOT_QSPI_XIP_IMAGE_NUMBER} CACHE INTERNAL "QSPI XIP image number" FORCE)
+
+        if(SB_CONFIG_MCUBOOT_BUILD_DIRECT_XIP_VARIANT)
+          set(mcuboot_secondary_app_QSPI_XIP_IMAGE_NUMBER ${SB_CONFIG_MCUBOOT_QSPI_XIP_IMAGE_NUMBER} CACHE INTERNAL "QSPI XIP image number" FORCE)
+        endif()
+
+        # Flash drivers are mandatory with QSPI XIP split image support
+        set_config_bool(${DEFAULT_IMAGE} CONFIG_FLASH y)
+        set_config_bool(mcuboot CONFIG_FLASH y)
+        set_config_bool(mcuboot CONFIG_MULTITHREADING y)
+        set_config_bool(mcuboot CONFIG_MCUBOOT_VERIFY_IMG_ADDRESS n)
+
+        if(NOT DEFINED SB_CONFIG_PM_OVERRIDE_EXTERNAL_DRIVER_CHECK)
```

Comment:
we have only QSPI XPI capability in our SoCs - this is for future non-existing targets

Reply:
I mean -dead code/scripts nowadays

Reply:
This is for forked drivers with modifications, which are a very real thing